clients/tests/pom.xml

```diff
@@ -48,13 +48,6 @@
       <version>${spark3.version}</version>
       <scope>provided</scope>
     </dependency>
-    <dependency>
-      <groupId>org.projectnessie</groupId>
-      <artifactId>nessie-client</artifactId>
-      <version>${project.version}</version>
-      <scope>provided</scope>
```

Comment:
Why doesn't provided or optional work? Are the upcoming changes to the client so drastic that we have to add this code duplication? why can't the clients/tests module import the `iceberg.nessie.version` of the client instead of this refactor?

Reply:
the idea is that delta for example might end up having a different version of the `nessie-client` than iceberg or not? That's why it seems safer to move the `nessie-client` to where it's actually required (into the subprojects)

Reply:
so technically speaking we could then also have a `deltalake.nessie.version` which is being used in the delta projects, thus making our lives easier in case Delta&Iceberg stuff uses different nessie versions